Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -819,7 +872,33 @@ void OnServerErrorInternal(NetworkConnection conn, ErrorMessage msg)
 
         #region Client Internal Message Handlers
 
-        void OnClientConnectInternal(NetworkConnection conn, ConnectMessage message)
+        void OnClientAuthenticateInternal(NetworkConnection conn, ConnectMessage message)
+        {
+            if (LogFilter.Debug) Debug.Log("NetworkManager.OnClientAuthenticateInternal");
+
+            if (authenticator != null)
+            {
+                // if we have an authenticator then let it handle authentication
+                authenticator.OnClientAuthenticateInternal(conn);
+            }
+            else
+            {
+                // otherwise authenticate immediately
+                OnClientAuthenticated(conn);
+            }
+        }
+
+        // called after successful authentication
+        void OnClientAuthenticated(NetworkConnection conn)
+        {
+            // set connection to authenticated
+            conn.isAuthenticated = true;
+
+            // proceed with the login handshake by calling OnClientConnect
+            OnClientConnectInternal(conn);
+        }
+
+        void OnClientConnectInternal(NetworkConnection conn)
```

Comment:
Move this code inside OnClientAuthenticated

Reply:
Can't - it'll break things

Reply:
how?   this method is private,  nobody is using it,  you can delete the method no problem without breaking anyone's code.